app/src/main/java/org/p2p/wallet/deeplinks/AppDeeplinksManager.kt

```diff
@@ -24,13 +28,56 @@ class AppDeeplinksManager(
         const val NOTIFICATION_TYPE = "eventType"
     }
 
+    private val deeplinkData = MutableStateFlow<DeeplinkData?>(null)
+
+    /**
+     * Tabs switcher for navigate by home screen (see [org.p2p.wallet.home.MainFragment])
+     */
     private var mainTabsSwitcher: MainTabsSwitcher? = null
+
+    /**
+     * Listener for root activity (see [org.p2p.wallet.root.RootActivity])
+     */
     private var rootListener: RootListener? = null
 
+    /**
+     * Intent for pending home intents, like attempt to
+     */
     private var pendingIntent: Intent? = null
 
+    /**
+     * Intent for send-via-link
+     */
     private var pendingTransferLink: SendViaLinkWrapper? = null
 
+    /**
+     * Notify manager that we have new deeplink to handle
+     * @param data Deeplink data
+     */
+    fun notify(data: DeeplinkData) {
+        deeplinkData.tryEmit(data)
+    }
+
+    /**
+     * Get the deeplink data flow for specified targets
+     * @param supportedTargets Set of deeplink targets, empty means all targets will be emitted
+     */
+    fun subscribeOnDeeplinks(supportedTargets: Set<DeeplinkTarget> = emptySet()): Flow<DeeplinkData> {
+        return deeplinkData
+            .filterNotNull()
+            .let { flow ->
```

Comment:
i think just `it` is okay here

Reply:
this statement has nested lambda, according to the guidelines it's preferrable to use explicit argument names in such cases